src/Elastic.Apm/Report/Serialization/BodyStringSanitizerConverter.cs

```diff
@@ -22,18 +22,23 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
 		{
 			if (value is string strValue)
 			{
-				var formValues = strValue.Split('&');
-
 				if (!FormatCheck(strValue))
 					writer.WriteValue(strValue);
 				else
 				{
+					var formValues = strValue.Split('&');
+
 					var sb = new StringBuilder();
 					foreach (var formValue in formValues)
 					{
 						var formsValueSplit = formValue.Split('=');
-						if (formsValueSplit.Length != 2)
-							continue;
+						if (formsValueSplit.Length != 2 || string.IsNullOrWhiteSpace(formsValueSplit[0])
+							|| string.IsNullOrWhiteSpace(formsValueSplit[1]))
+						{
+							//in this case it's not in the expected format, so we won't (and can't) sanitize, we just write it and move on
+							writer.WriteValue(strValue);
```

Comment:
After this change `FormatCheck` becomes redundant - making it always return `true` doesn't cause any tests to fail...

Reply:
Well, it is not so easy.

Here is my thinking:
Iterating through a string is very fast - and we can find formats that we can't sanitize with an iteration, but as it turned out, it's not 100% correct.

On the other hand after the `FormatCheck` and line 39 (where we'd end up without `FormatCheck`) there is an allocation, which is very very bad. 

To show this, here is a benchmark I wrote:

```
[Benchmark]
public void Sanitizer()
{
	var converter = new BodyStringSanitizerConverter(new EnvironmentConfigurationReader());
	var sb = new StringBuilder();
	for (var i = 0; i < 10000; i++)
	{
		sb.Append("aa");
	}
	sb.Append("=bb&&c=dd");
	converter.WriteJson(new JsonTextWriter(new DebugTextWriter()), sb.ToString(), null );
}
```

Here is the result with the PR as it is (including `CheckFormat`):

```

|    Method |     Mean |    Error |   StdDev | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|---------- |---------:|---------:|---------:|------------:|------------:|------------:|--------------------:|
| Sanitizer | 98.58 us | 1.527 us | 1.428 us |     15.1367 |      0.1221 |           - |             93.2 KB |
```

And here is the result without the "redundant" `CheckGormat`):

```
|    Method |    Mean |    Error |   StdDev | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|---------- |--------:|---------:|---------:|------------:|------------:|------------:|--------------------:|
| Sanitizer | 2.182 s | 0.0148 s | 0.0131 s |           - |           - |           - |              5.1 MB |
```

As you see the difference is big.

You can argue that my sample data is kind of an edge case, but my point with this is that 1 additional iteration in `CheckFormat` adds very little overhead, while the additional allocation adds way more overhead.